core/src/main/java/io/grpc/Metadata.java

```diff
@@ -69,6 +69,27 @@
   public static final String BINARY_HEADER_SUFFIX = "-bin";
 
   /**
+   * Simple metadata marshaller that encodes bytes as is.
+   *
+   * <p>This should be used when raw bytes are favored over un-serialized version of object. Can be
+   * helpful in situations where more processing to bytes is needed on application side, avoids
+   * double encoding/decoding.</p>
+   */
+  public static final BinaryMarshaller<byte[]> BINARY_BYTE_MARSHALLER =
+      new BinaryMarshaller<byte[]>() {
+
+    @Override
+    public byte[] toBytes(byte[] value) {
+      return value;
```

Comment:
As discussed on IRC, how do you feel about making a copy of the byte[] here and in parseBytes? That seems like it will reduce chances of the application modifying the byte[] and corrupting the value stored/cached in Metadata.